web/api/v1/api.go

```diff
@@ -482,7 +482,18 @@ func returnAPIError(err error) *apiError {
 }
 
 func (api *API) labelNames(r *http.Request) apiFuncResult {
-	q, err := api.Queryable.Querier(r.Context(), math.MinInt64, math.MaxInt64)
+	start, err := parseTimeParam(r, "start", minTime)
```

Comment:
I did not look deeper but we need to make sure to not break compatibility in the same major version of Prometheus. We can only add things, we cannot suddenly fail all requests without min and max time for all tens of thousands users (((: 

Let's make sure we assume no start and end params works as previously. A test case for this would be amazing as well. 🤗

Reply:
Start and end are optional parameters so that compatibility should be maintained. If a start and end are not provided, the code defaults to how the code was operating previously. 

I believe there are a few tests that ensure this:
- https://github.com/prometheus/prometheus/pull/7288/files#diff-c32caf482fb76fa695ceffe3455c6d9aR1479-R1480?
- https://github.com/prometheus/prometheus/blob/957a57db5bb478d5723fe93d2678d674bce87e20/web/api/v1/api_test.go#L404

Let me know if you think I should add any other tests or if this is sufficient.

Reply:
No it's perfect, looks like this is done then (: